lib/galaxy/web_stack/__init__.py

```diff
@@ -222,8 +223,8 @@ def run_postfork(cls):
     def late_postfork(cls):
         # We can't run postfork functions immediately, because this is before the gunicorn `post_fork` hook runs,
         # and we depend on the `post_fork` hook to set a worker id.
-        t = threading.Thread(target=cls.run_postfork)
-        t.start()
+        cls.late_postfork_thread = threading.Thread(target=cls.run_postfork)
+        cls.late_postfork_thread.start()
```

Comment:
Should we join the thread here as well ?

Reply:
Not sure, I think it may defeat the purpose of having a separate thread.

Reply:
Oh yeah, it's actually in the comment above.